erpnext/agriculture/doctype/plant/plant.py

```diff
@@ -56,7 +56,12 @@ def update_plant(source, target):
 			}
 		}}, target_doc, update_plant)
 
-	return target_doc
+	completely_harvested = frappe.get_all("Harvest", {'plant':source_name, "is_finished":1}, "*")
```

Comment:
you can simply check self. completely_harvested

Reply:
changed condition